full-node/sov-prover-storage-manager/src/lib.rs

```diff
@@ -121,21 +156,35 @@ where
                 new_snapshot_id
             }
         };
+        tracing::debug!(
+            "Requested native storage for block {:?}, giving snapshot id={}",
+            block_header,
+            new_snapshot_id
+        );
 
+        self.get_storage_with_snapshot_id(new_snapshot_id)
+    }
+
+    fn get_finalized_storage(&mut self) -> anyhow::Result<Self::NativeStorage> {
```

Comment:
Why a `get` method has a `&mut self` and modify an internal state.

Reply:
Naming issue, in fact it creates storage, so I will rename it to "create"